perception/multi_object_tracker/src/tracker/model/bicycle_tracker.cpp

```diff
@@ -214,37 +214,35 @@ bool BicycleTracker::predict(const double dt, KalmanFilter & ekf) const
   return true;
 }
 
+/**
+ * @brief measurement update with ekf on receiving detected_object
+ *
+ * @param object : Detected object
+ * @return bool : measurement is updatable
+ */
 bool BicycleTracker::measureWithPose(
   const autoware_auto_perception_msgs::msg::DetectedObject & object)
 {
-  constexpr int dim_y = 2;  // pos x, pos y, depending on Pose output
-  // double measurement_yaw =
-  //   tier4_autoware_utils::normalizeRadian(tf2::getYaw(object.state.pose_covariance.pose.orientation));
-  // {
-  //   Eigen::MatrixXd X_t(ekf_params_.dim_x, 1);
-  //   ekf_.getX(X_t);
-  //   while (M_PI_2 <= X_t(IDX::YAW) - measurement_yaw) {
-  //     measurement_yaw = measurement_yaw + M_PI;
-  //   }
-  //   while (M_PI_2 <= measurement_yaw - X_t(IDX::YAW)) {
-  //     measurement_yaw = measurement_yaw - M_PI;
-  //   }
-  //   float theta = std::acos(
-  //     std::cos(X_t(IDX::YAW)) * std::cos(measurement_yaw) +
-  //     std::sin(X_t(IDX::YAW)) * std::sin(measurement_yaw));
-  //   if (tier4_autoware_utils::deg2rad(60) < std::fabs(theta)) return false;
-  // }
+  bool use_orientation_information = false;
+  // if orientation is available
+  if (object.kinematics.orientation_availability) {
```

Comment:
I think orientation_availability should be checked by SIGN_UNKNOWN or AVAILABLE.
https://github.com/tier4/autoware_auto_msgs/blob/tier4/main/autoware_auto_perception_msgs/msg/DetectedObjectKinematics.idl#L21-L26

Reply:
OK. Then, how do you think about adding threshold to check if the predicted and measured yaw angle are close enough?
I think 

- Constant threshold $\theta_d$
- Thresholds that take into account the covariance of the estimation, such as mahalanobis distance
- No threshold. just feed any measurements to EKF (choose this if raw detection is much more stable than tracking)